### PR TITLE
Tls Config for Messaging extensions

### LINKED ISSUE #41503: Allows the Kafka client to be configured using the TLS registry
### Description

Allows the Kafka client to be configured with the TLS registry (for the TLS aspect of it).

Basically, when `kafka.tls-configuration-name` is set, we need to locate the TLS configuration from the registry and set:

```
security.protocol=SSL
ssl.truststore.location=/var/private/ssl/kafka.client.truststore.jks  
ssl.truststore.password=test1234. # Trust store password
ssl.keystore.location=/var/private/ssl/kafka.client.keystore.jks  
ssl.keystore.password=test1234 # Key store password
ssl.key.password=test1234 # Alias password
```

Unfortunately, the KAfka configuration only accepts files. This means we would need to extract these files from the configuration and cannot use the already processed KeyStore instances.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by ozangunalp**
Includes refactoring for common TLS configuration of Vert.x-based clients.

Closes #41503 and #41504.

To merge after the final Reactive messaging 4.24.0 release

### CODE REVIEW

**Comment by ozangunalp** [comment]
Those tests became flaky very recently.

File: extensions/kafka-client/runtime/pom.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-caffeine</artifactId>
```
Comment (cescoffier) [inline extensions/kafka-client/runtime/pom.xml:48]: Why did we have caffeine as dependency?

File: extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/tls/QuarkusKafkaSslEngineFactory.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Override
+    public boolean shouldBeRebuilt(Map<String, Object> nextConfigs) {
+        return false;
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Set.of();
```
Comment (cescoffier) [inline extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/tls/QuarkusKafkaSslEngineFactory.java:51]: What is the idea being this method? Is this about certificate renewal ?

File: extensions/smallrye-reactive-messaging-kafka/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/KafkaConfigCustomizer.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import io.quarkus.kafka.client.tls.QuarkusKafkaSslEngineFactory;
+import io.smallrye.reactive.messaging.ClientCustomizer;
+
+@ApplicationScoped
+public class KafkaConfigCustomizer implements ClientCustomizer<Map<String, Object>> {
+
+    @Override
+    public Map<String, Object> customize(String channel, Config channelConfig, Map<String, Object> config) {
+        // TODO verify other ssl properties
```
Comment (cescoffier) [inline extensions/smallrye-reactive-messaging-kafka/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/KafkaConfigCustomizer.java:17]: What do you mean?

**Review by cescoffier — APPROVED** [review]
Awesome!

File: extensions/kafka-client/runtime/pom.xml (hunk 1 above)
Comment (ozangunalp) [reply extensions/kafka-client/runtime/pom.xml:48]: It was used for caching Subjets in SubjectSubstitution but then dropped completely.

File: extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/tls/QuarkusKafkaSslEngineFactory.java (hunk 2 above)
Comment (ozangunalp) [reply extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/tls/QuarkusKafkaSslEngineFactory.java:51]: If I understood correctly those are for dynamically reconfigurable server options. 
For the client, I didn't see any certificate renewal mechanism.

File: extensions/smallrye-reactive-messaging-kafka/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/KafkaConfigCustomizer.java (hunk 3 above)
Comment (ozangunalp) [reply extensions/smallrye-reactive-messaging-kafka/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/KafkaConfigCustomizer.java:17]: If other ssl-related Kafka client config has been provided, by configuring the SslEngineFactory we'd simply not use them. If we detect SSL configs in addition to the tls-configuration-name we could log a warning.

**Pull request merged**